### ISSUE TITLE
Video Block: Add accessibility name to core/video block using a caption

### ISSUE THREAD

**Issue by shubhtoy** [WordPress/gutenberg#70297]
### Description

The following issue was raised on trac: [#63519](https://core.trac.wordpress.org/ticket/63519) by [@rlautan](https://profiles.wordpress.org/rlautan/)

The `core/video` block that uses a caption has no accesibility name. Adding one as a aria-label or a `aria-labbeledby` attribute will enhance the accessibility for the block.

### Step-by-step reproduction instructions

1. Create a new page/post
2. Add a `core/video` block
3. Add media and caption to it
4. Navigate to the frontend, and using a screenreader try to focus on the video.
5. Verify that the screenreader announces the video but not the caption

### Screenshots, screen recording, code snippet

https://github.com/user-attachments/assets/90045664-b899-4b55-b8e0-cbdbb2a8791d

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by shubhtoy** [comment]
I would love to work on this, please let me know if I can go ahead an add a `aria-labelledby` field to the video tag for a11y.

**Comment by t-hamano** [comment]
Thanks for the report.

I think the HTML improvements suggested here are something like this:

```html
<!-- Current -->
<figure class="wp-block-video">
	<video src="sample.mp4"></video>
	<figcaption class="wp-element-caption">Video Caption</figcaption>
</figure>

<!-- Proposed -->
<figure class="wp-block-video">
	<video src="sample.mp4" aria-labelledby="video-caption"></video>
	<figcaption class="wp-element-caption" id="video-caption">Video Caption</figcaption>
</figure>
```

While this approach does allow screen readers to read the caption when the video element is focused, it also feels a bit overkill.

Below are two sources for making the video element accessible, but neither source mentions an approach using labels.

- [Video element visual content has accessible alternative[proposed] | ACT Rule | WAI | W3C](https://www.w3.org/WAI/standards-guidelines/act/rules/c5a4ea/proposed/)
- [Video element visual-only content has accessible alternative[proposed] | ACT Rule | WAI | W3C](https://www.w3.org/WAI/standards-guidelines/act/rules/c3232f/proposed/)

So I'm not sure if the current HTML is truly inaccessible, or what the ideal approach would be.

cc @joedolson @afercia @alexstine @amberhinds

**Comment by shubhtoy** [comment]
Thanks for your humble input @t-hamano , I will wait for further clarification on the same, and move forward accordingly.

**Comment by joedolson** [comment]
`figcaption` is intended to label the media figure, so it's read when the figure is reached by the screen reader. It's not a label for the video itself, per se. I don't think that adding a second labeling context is a good idea, because it would result in announcing that label twice.  

It is good to provide a label of some kind for a video element; if there's no figcaption provided, then the video is unlabeled. That's easily resolved by adding a heading, though. 

In my opinion, it wouldn't be a help for this labeling to be repeated for the video. Adding support for an `aria-label` for the video  when it doesn't have a caption would be potentially worthwhile, however.

**Comment by shubhtoy** [comment]
Thanks @joedolson , this makes sense, I will add an aria-label field conditionally based on presence of a video caption.

**Comment by shubhtoy** [comment]
Hey @t-hamano @joedolson , I've opened a PR with suggested changes, please review it your convenience. 

Thanks!

**Comment by t-hamano** [comment]
@shubhtoy Unfortunately, #70364 does not solve anything, so I have closed it.

> Adding support for an `aria-label` for the video  when it doesn't have a caption would be potentially worthwhile

As I understand it, one approach would be to add a text field to the block sidebar that allows the user to enter a description of the video, and then map it as the value of the aria-label attribute on the video element.

However, I question whether this UI is really necessary - as noted later in [this comment](https://github.com/WordPress/gutenberg/issues/70297#issuecomment-2947390971), there are various ways to make video elements accessible. Most of the approaches proposed there should be achievable with the current block editor.

**Comment by shubhtoy** [comment]
Thank you for the clarification, @t-hamano. I understand now that using aria-labelledby is effective with screen readers. Would it be alright if I proceed with this approach and open a PR accordingly?

**Comment by t-hamano** [comment]
> However, I question whether this UI is really necessary - as noted later in https://github.com/WordPress/gutenberg/issues/70297#issuecomment-2947390971, there are various ways to make video elements accessible. Most of the approaches proposed there should be achievable with the current block editor.

For the reasons above, I'm not sure that a UI for aria-label is absolutely necessary. I'll hear what others think.

If we decide to implement a UI for aria-label, note the following two points.

- It is a good idea to use the `arialabel` block support.
- To prevent redundant reading, we should disable the aria-label attribute if the block has figcaption text.

**Comment by amberhinds** [comment]
I don't think that the aria-label should be conditionally referenced. The label a.k.a. accessible name of a video, should be short and concise. A caption can be quite long and used to convey other information.

Why not use the title from the Media Library as the aria-label? That doesn't require a new field on the block. If we're worried that videos will have random  rather than purposefully edited titles, then I'd add a "Video Title" field in the block that initially fills based on the Media Library title but is editable on the block level as alt text is.

**Comment by sonawaneapekshit** [comment]
Agreed to what @amberhinds suggested in above comment i.e. using media's title to be used as aria-label. we can either use title or description  form media library of the video as aria-label and have figcaption as fallbackl. But need to be mindful about what is allowed word length for it.

@t-hamano Could you please provide your inputs on this?

Priority:
1. Use media title as aria-label
2. If title is missing → fallback to figcaption

**Comment by t-hamano** [comment]
It's easy to add an aria-label to a video element, but a video block can also display subtitles and captions. Is the following HTML problematic from an accessibility perspective?

```html
<figure class="wp-block-video">
  <video controls src="video.mp4" aria-label="Video label">
    <track src="test.vtt" label="test" srclang="en" kind="subtitles" />
  </video>
  <figcaption class="wp-element-caption">Caption</figcaption>
</figure>
```

**Comment by sonawaneapekshit** [comment]
I checked the documentation regarding the priority for the aria-label is given higher priority if present 

<img width="1348" height="388" alt="Image" src="https://github.com/user-attachments/assets/ae8f4ec2-41cd-4a96-82a8-26a4f9a7c598" />

https://www.w3.org/TR/html-aapi/#accessible-name-and-description-calculation
(screenshot from the link about figure and figcaption)

**Comment by joedolson** [comment]
An `aria-label` on the `figure` would have higher priority, but the example @t-hamano showed has the `aria-label` on the `video` element, which is a different situation. 

It's essentially equivalent to using an `alt` attribute on an image inside a figure; there's a label that describes the figure (the `figcaption`) and there's a label that describes the media (the `aria-label`).

Having an `aria-label` on the video is fine; I think the only important thing we'd want to be cautious about is preventing duplication - the `aria-label` and the `figcaption` should not contain the same text, and only one of the two needs to be required.
